### ISSUE TITLE
[Due for payment 2025-02-18]  Trip - Room name field does not include left parenthesis of the trip room name

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54539]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp https://github.com/Expensify/App/pull/52823
**Email or phone of affected tester (no customers):** applausetester+0948501@applause.expensifail.com
**Issue reported by:** Applause Internal Team
**Device used:** Windows 11
**App Component:** Other

## Action Performed:
Precondition:
- Create a trip with steps from this PR https://github.com/Expensify/App/pull/52823

1. Go to staging.new.expensify.com
2. Go to the trip room.
3. Click on the room header
4. Click Room name.

## Expected Result:
Room name field should contain the complete room name.

## Actual Result:
Room name field does not include the first character of the room name, which is the left parenthesis.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/e832a09c-9ae8-4a4d-875e-dc3a1416dc23

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

**Comment by laurenreidexpensify** [comment]
I see latest version of staging is v9.0.79-4, so let's retest on that before moving forward

**Comment by laurenreidexpensify** [comment]
Waiting for retest

**Comment by CyberAndrii** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Trip - Room name field does not include left parenthesis of the trip room name

### What is the root cause of that problem?

In code, the room name is assumed to always be prefixed with `#`. However, since trip rooms are not prefixed, the first character is removed instead of specifically removing `#`.

https://github.com/Expensify/App/blob/1cfcfa47ca046760f363114c10b1d77999147f9b/src/components/RoomNameInput/index.tsx#L57

### What changes do you think we should make in order to solve the problem?

Either prefix trip room names with `#` or modify the code to remove the first character only if the name starts with `#`

```diff
-value={value?.substring(1)}
+value={value?.startsWith(CONST.POLICY.ROOM_PREFIX) ? value.substring(1) : value}
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

A test to validate the value of `TextInput` with and without the `#` prefix.

**Comment by laurenreidexpensify** [comment]
@blazejkustra pls comment on issue so I can assign you thanks

**Comment by blazejkustra** [comment]
Sorry, I was OOO until today. Commenting for assignment @laurenreidexpensify

**Comment by blazejkustra** [comment]
I'm a bit confused, it seems like I can't edit the room name for my trips:

https://github.com/user-attachments/assets/ad97c941-f4b1-410d-b073-ac5800c25f6f

Could it be that something changed during last two weeks and now editing the name is disabled? @IuliiaHerets Could you retest this or provide more detailed steps? cc @laurenreidexpensify

**Comment by blazejkustra** [comment]
@IuliiaHerets Could you add me to that workspace and make me an admin? That could make it easier to debug. Here is my email: `blazej.kustra+trav3@swmansion.com`

**Comment by stitesExpensify** [comment]
Chatted 1:1 with @blazejkustra , we think we might be able to just remove the restriction

**Comment by blazejkustra** [comment]
There’s one potential approach we could explore, but it would require some backend changes. While chats currently have the naming restriction described earlier (only allowing lowercase letters, numbers, and hyphens), group chats do not. If we reuse the group chat logic for trip rooms, we could potentially support this functionality without naming restrictions. Here is how it would work:


https://github.com/user-attachments/assets/cb411313-fe8a-4a4c-9a88-f03fdce76a61


That said, there’s one blocker to this approach. Under the hood `UpdateGroupChatName` command is used. Currently it returns an error when attempting to use it for a trip room. Addressing this would require adjustments to how the backend handles trip room name updates. 

cc @stitesExpensify @rlinoz

**Comment by rlinoz** [comment]
Looking at the backend code I think we should create a `UpdateTripRoomName` command

**Comment by stitesExpensify** [comment]
So to clarify are we already calling `UpdateGroupChatName` for both regular trips and groups currently?

**Comment by blazejkustra** [comment]
Currently trip chats are usually threads, and you can't change the name:
![image](https://github.com/user-attachments/assets/02fa62c5-a836-41bf-ae5d-410209abd0a2)

This edge case occurs when creating a trip from olddot in a workspace which doesn't have the workspace chat (this way the trip isn't a thread), under the hood it uses `UpdatePolicyRoomName` now:

![image](https://github.com/user-attachments/assets/13af1489-2a88-4b43-be37-71a26ebf5cc1)

**Comment by blazejkustra** [comment]
@rlinoz @stitesExpensify Should we proceed with the backend change?

**Comment by rlinoz** [comment]
also we can't `UpdateGroupChatName` checks the `chatType` and that would fail. I think the new command best fits _the one command per user action_ guideline.

**Comment by blazejkustra** [comment]
Makes sense to me, it'll be a small change on FE if we have a new command for it 👍

**Comment by rlinoz** [comment]
@stitesExpensify if you agree on the new command I can create it if your plate is full

**Comment by stitesExpensify** [comment]
Hey @rlinoz that would be great 🙇

**Comment by laurenreidexpensify** [comment]
Just catching up here - does https://github.com/Expensify/Auth/pull/13903 fix this?

**Comment by rlinoz** [comment]
Nope, we are still missing a Web-E PR that I will put for review today

**Comment by rlinoz** [comment]
Command should be deployed to staging today.

**Comment by blazejkustra** [comment]
Update: I had other work today and haven't finished this one yet. I'll raise a PR tomorrow, if all works well with connecting to backend this should be ready for review tomorrow as well 😄

**Comment by blazejkustra** [comment]
Update: [PR](https://github.com/Expensify/App/pull/55936) is up! @shubham1206agra could you have a look?

**Comment by laurenreidexpensify** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by laurenreidexpensify** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~021891861354300781002 for payment @shubham1206agra pls apply

**Comment by laurenreidexpensify** [comment]
Contract sent to Shubham in Upwork

**Comment by laurenreidexpensify** [comment]
@shubham1206agra bump thanks

**Comment by laurenreidexpensify** [comment]
Spoke to Shubnam about this, downgrading to monthly for now

**Comment by stitesExpensify** [comment]
What is the latest on this payment? Can we close the issue?

**Comment by laurenreidexpensify** [comment]
@shubham1206agra offer sent ^^

**Comment by laurenreidexpensify** [comment]
## Payment Summary

C+ Reviewer @shubham1206agra paid $250 in Upwork
